Entities/src/main/java/tw/waterball/judgegirl/entities/stubs/SubmissionStubBuilder.java

```diff
@@ -82,11 +80,11 @@ public String getId() {
 
     @Override
     public Optional<Verdict> mayHaveVerdict() {
-        return judges.isEmpty() ? Optional.empty() : Optional.of(new Verdict(judges));
+        return verdict.mayHaveVerdict();
```

Comment:
`return verdictBuilder == null ? Optional.empty() : Optional.of(verdictBuilder.build());`

To achieve this, you should lazy-initialize the `verdictBuilder` in each judge-status's construction method.